students/TinaB/lesson04/mailroom_part2.py

```diff
@@ -0,0 +1,231 @@
+#!/usr/bin/env python3
+"""
+$ chmod +x mailroom_part2.py
+"""
+import datetime
+import os
+
+# list of donors
+donors_list_dictionary = [{"firstname": "William", "lastname": "Gates", "donations": [326892.24, 122, 22, 12], "donate_times": [4]},
+                          {"firstname": "Mark", "lastname": "Zuckerberg", "donations": [
+                              30, 60, 65982.55], "donate_times": [3]},
+                          {"firstname": "Jeff", "lastname": "Bezos",
+                              "donations": [52636.27], "donate_times": [1]},
+                          {"firstname": "Paul", "lastname": "Allen",
+                              "donations": [877.33, 22], "donate_times": [2]},
+                          {"firstname": "Steven", "lastname": "Hawking", "donations": [
+                              326892.24, 123, 123.33, 123, 123], "donate_times": [5]},
+                          {"firstname": "Justin", "lastname": "Timberlake", "donations": [999658.25, 1233, 123], "donate_times": [3]}]
+
+# Menus
+def menu_selection(prompt, dispatch_dict):
+    while True:  # loop forever until they quit
+        response = input(prompt)
+        if dispatch_dict[response]() == "exit menu":
+            break
+
+
+def single_print_sub_menu():
+    menu_selection(single_print_sub_prompt, single_print_sub_dispatch)
+
+
+def quit():
+    print("Quitting this menu now")
+    return "exit menu"
+
+# functions for prompts ----
+
+
+def get_name_input():
+    # Function to select user input to return to print function
+    user_input = input("\nLookup donor by: \n"
+                       "\t1. Firstname: \n"
+                       "\t2. Lastname:\n"
+                       "Please enter 1 or 2 >>  ")
+
+    donor_return = []
+    donor_check = 0
+    if int(user_input) == 1:
+        fname = input(
+            "Type the first name of the donor you are looking for:  ")
+    else:
```

Comment:
Probably better to go with `elif user_input == '2'` here, since this will grab any input other than '1'. You could have an additional `else` clause to handle bad input